tests/test_interaction_kernel.py

```diff
@@ -0,0 +1,144 @@
+import numpy as np
+import pytest
+
+from parcels import (
+    FieldSet, ParticleSet, JITParticle, StateCode
+)
+from parcels.particle import ScipyInteractionParticle, Variable
+from parcels.application_kernels.interaction import NearestNeighborWithinRange,\
+    AsymmetricAttraction
+from parcels.application_kernels.interaction import MergeWithNearestNeighbor
+
+ptype = {'scipy': ScipyInteractionParticle, 'jit': JITParticle}
+
+
+def DummyMoveNeighbor(particle, fieldset, time, neighbors, mutator):
+    """A particle boosts the movement of its nearest neighbor, by adding
+    0.1 to its lat position.
+    """
+    if len(neighbors) == 0:
+        return StateCode.Success
+
+    distances = [np.sqrt(n.vert_dist**2 + n.horiz_dist**2) for n in neighbors]
+    i_min_dist = np.argmin(distances)
+
+    def f(p):
+        p.lat += 0.1
+
+    neighbor_id = neighbors[i_min_dist].id
+    mutator[neighbor_id].append((f, ()))
+
+    return StateCode.Success
+
+
+def DoNothing(particle, fieldset, time):
+    return StateCode.Success
+
+
+def fieldset(xdim=20, ydim=20):
+    """ Standard unit mesh fieldset """
+    lon = np.linspace(0., 1., xdim, dtype=np.float32)
+    lat = np.linspace(0., 1., ydim, dtype=np.float32)
+    U, V = np.meshgrid(lat, lon)
+    data = {'U': np.array(U, dtype=np.float32), 'V': np.array(V, dtype=np.float32)}
+    dimensions = {'lat': lat, 'lon': lon}
+    return FieldSet.from_data(data, dimensions, mesh='spherical')
+
+
+class MergeParticle(ScipyInteractionParticle):
+    nearest_neighbor = Variable('nearest_neighbor', dtype=np.int64, to_write=False)
+    mass = Variable('mass', initial=1, dtype=np.float32)
+
+
+@pytest.fixture(name="fieldset")
+def fieldset_fixture(xdim=20, ydim=20):
+    return fieldset(xdim=xdim, ydim=ydim)
+
```

Comment:
Because of the changes in the main-loop with time, it is very much advisable to test forward- and backward advection and long-time small-timestep runs with the new kernels.

Reply:
I have added a backward time test for the last (dummy) test. The other interaction kernels are generally not time reversible (especially the merge kernel would be hard to define in reversed time). Let me know if you had a more elaborate scheme in mind.